contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/GobblinDistcpJob.java

```diff
@@ -0,0 +1,197 @@
+            info("          End of Distcp");
+            info("####################################");
+        }
+    }
+
+    private void runDistcp(Path from, Path to) throws Exception {
+        info("source: " + from);
+        info("CDN: " + to);
+        EmbeddedGobblin embeddedGobblin = new EmbeddedGobblinDistcp(from, to).mrMode();
+
+        // Used for global throttling"
+        embeddedGobblin.distributeJar("lib/*");
+
+        for (Map.Entry<String, String> entry : this.props.entrySet()) {
+            if (entry.getKey() != null && (entry.getKey()).startsWith(ATTR_PREFIX)) {
+                String key = (entry.getKey()).substring(ATTR_PREFIX.length());
+                embeddedGobblin.setConfiguration(key, entry.getValue());
+            }
+        }
+        JobExecutionResult result =  embeddedGobblin.run();
+
+        if (!result.isSuccessful()) {
+            throw new RuntimeException("Distcp job failed!", result.getErrorCause());
+        }
+    }
+
+    private FileSystem getRootFS(String target) throws Exception {
+        // Remove directory path from URL, e.g. hdfs://hostname:9000/a/b/c => hdfs://hostname:9000
+        URI rootURI = new URI(target.replaceAll("(?<=:[0-9]{1,5})/.*", ""));
+        return FileSystem.get(rootURI, new JobConf());
+    }
+
+    private void deleteDir(FileSystem fs, String target) throws Exception {
+        // Extract directory path from URL, e.g. hdfs://hostname:9000/a/b/c => /a/b/c
+        Path path = new Path(target.replaceAll(".*://.*?(?=/)", ""));
+
+        if (fs.exists(path)) {
+            fs.delete(path, true);
+            if (fs.exists(path)) {
+                warn("Could not delete temp directory " + path + " in CDN!");
+            } else {
+                info("Deleted " + path);
+            }
+        }
+    }
+
+    private void deleteDirOnExit(FileSystem fs, String target) throws Exception {
+        // Extract directory path from URL, e.g. hdfs://hostname:9000/a/b/c => /a/b/c
+        Path path = new Path(target.replaceAll(".*://.*?(?=/)", ""));
+
+        fs.deleteOnExit(path);  // Delete the directory even if an exception occurs
+        info(path + " is scheduled to be deleted on exit.");
+    }
+
+    private void addPermissionsToParents(FileSystem fs, String target) throws Exception {
+        // Extract directory path from URL, e.g. hdfs://hostname:9000/a/b/c => a/b/c
+        String pathFromRoot = target.replaceAll(".*://.*?/", "");
+        String parent = "";
+
+        for (String seg: pathFromRoot.split("/")) {
+            parent = parent + "/" + seg;
+            Path parentPath = new Path(parent);
+            FsPermission perm = fs.getFileStatus(parentPath).getPermission();
+            FsAction u = perm.getUserAction();
+            FsAction g = perm.getGroupAction();
+            FsAction o = perm.getOtherAction();
+            boolean changed = false;
+
+            if (props.getBoolean(VoldemortBuildAndPushJob.PUSH_CDN_READBYGROUP, false)) {
```

Comment:
Can we have underscores between each and every word? My eyes tripped on this a few times, parsing "ready", "baby", and other non-sense :sweat_smile: instead of the intended meaning.